### ISSUE TITLE
tf.image.extract_glimpse crashes with negative input

### ISSUE THREAD

**Issue by lugalUrim** [tensorflow/tensorflow#51618]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.4
- Python version: 3.6

**Standalone code to reproduce the issue**
```
import tensorflow as tf
import numpy as np
x = np.arange(9).reshape([1,3,3,1])
res = tf.image.extract_glimpse(x, size=[1023, -63], offsets=[1023, 63], centered=False, normalized=False) # Crash
```

**Describe the current behavior**

It crashes when I execute the above code.

**Describe the expected behavior**
Should throw a `ValueError`.

**Comment by LudwigStumpp** [comment]
Happy to add the `app.log` from Google Colab executing above snippet for the latest version `2.6.0`.

Seems to me like the call of `CHECK_LT()` in `tensorflow/core/framework/tensor_shape.cc:569` defined at `tensorflow/tensorflow/tensorflow/core/platform/default/logging.h:413` causes that issue.

However, I am not a C++ expert, so I don't get any much further than that for now.

<html><body>


Timestamp | Level | Message
-- | -- | --
Aug 22, 2021, 12:34:44 AM | WARNING | WARNING:root:kernel 8e337399-7e5c-4782-aaa6-34f661e5c692 restarted
Aug 22, 2021, 12:34:44 AM | INFO | KernelRestarter: restarting kernel (1/5), keep random ports
Aug 22, 2021, 12:34:44 AM | WARNING | 2021-08-21 22:34:44.441658: F tensorflow/core/framework/tensor_shape.cc:569] Check failed: size >= 0 (-63 vs. 0)
Aug 22, 2021, 12:34:44 AM | WARNING | 2021-08-21  22:34:44.376372: I  tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver  does not appear to be running on this host (c679a9bf8bfa):  /proc/driver/nvidia/version does not exist
Aug 22, 2021, 12:34:44 AM | WARNING | 2021-08-21  22:34:44.375437: E tensorflow/stream_executor/cuda/cuda_driver.cc:271]  failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is  detected
Aug 22, 2021, 12:34:37 AM | INFO | Adapting to protocol v5.1 for kernel 8e337399-7e5c-4782-aaa6-34f661e5c692
Aug 22, 2021, 12:34:35 AM | INFO | Kernel started: 8e337399-7e5c-4782-aaa6-34f661e5c692
Aug 22, 2021, 12:33:19 AM | INFO | Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
Aug 22, 2021, 12:33:19 AM | INFO | http://172.28.0.12:9000/
Aug 22, 2021, 12:33:19 AM | INFO | The Jupyter Notebook is running at:
Aug 22, 2021, 12:33:19 AM | INFO | Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
Aug 22, 2021, 12:33:19 AM | INFO | http://172.28.0.2:9000/
Aug 22, 2021, 12:33:19 AM | INFO | The Jupyter Notebook is running at:
Aug 22, 2021, 12:33:19 AM | INFO | 0 active kernels
Aug 22, 2021, 12:33:19 AM | INFO | Serving notebooks from local directory: /
Aug 22, 2021, 12:33:19 AM | INFO | 0 active kernels
Aug 22, 2021, 12:33:19 AM | INFO | Serving notebooks from local directory: /
Aug 22, 2021, 12:33:19 AM | INFO | google.colab serverextension initialized.
Aug 22, 2021, 12:33:19 AM | INFO | google.colab serverextension initialized.
Aug 22, 2021, 12:33:19 AM | INFO | Writing notebook server cookie secret to /root/.local/share/jupyter/runtime/notebook_cookie_secret
Aug 22, 2021, 12:33:19 AM | INFO | Writing notebook server cookie secret to /root/.local/share/jupyter/runtime/notebook_cookie_secret


</body>
</html>

**Comment by sushreebarsa** [comment]
@lugalUrim Could you please refer to the [link](https://www.tensorflow.org/api_docs/python/tf/image/extract_glimpse) . Please try to upgrade the TF version 2.4 to 2.6.0 and refer to the above [comment](https://github.com/tensorflow/tensorflow/issues/51618#issuecomment-903185385) .Please let us know if it helps ?

**Comment by yongtang** [comment]
Added a PR #51618 for the fix.

**Comment by sushreebarsa** [comment]
@lugalUrim This issue will be closed once the PR is merged! Thanks!

**Comment by oawxkw** [comment]
I also observed the following API aliases or similar APIs can cause the same issue in older versions of tensorflow.
Users should be cautious when using them on both CPU and GPU up to tensorflow 2.7.4 (v2.7.3-139-ga73cc22ba39) except for tensorflow 2.6.1 (v2.6.0-101-g3aa40c3ce9d).

- `tf.image.extract_glimpse`, `tf.compat.v1.image.extract_glimpse`

It seems to be fixed in tensorflow 2.8.0 (v2.8.0-rc1-32-g3f878cff5b6) and later versions.
